src/state/accountSelectors.ts

```diff
@@ -466,6 +466,25 @@ export const getCurrentMarketFills = createAppSelector(
     !currentMarketId ? [] : marketFills[currentMarketId]
 );
 
+const getFillsForOrderId = createAppSelector(
+  [(s, orderId) => orderId, getSubaccountFills],
+  (orderId, fills) => (orderId ? groupBy(fills, 'orderId')[orderId] ?? [] : [])
+);
+
+/**
+ * @returns the average price the order is filled at
+ */
+export const getAverageFillPriceForOrder = () =>
+  createAppSelector([(s, orderId) => getFillsForOrderId(s, orderId)], (fillsForOrderId) => {
+    let total = 0;
+    let totalSize = 0;
+    fillsForOrderId.forEach((fill) => {
+      total += fill.price * fill.size;
+      totalSize += fill.size;
```

Comment:
if not, you probably want to apply some rounding to the value since js multiplication can sometimes expand the number of decimal places unexpectedly, like here's the average of two fills across one order: 
<img width="329" alt="image" src="https://github.com/user-attachments/assets/85af30d0-ae8f-4850-a1b9-143ce0517ba0">